lib/ProductOpener/Display.pm

```diff
@@ -7066,25 +7066,23 @@ sub display_blocks($)
 	my $blocks_ref = $request_ref->{blocks_ref};
 
 	my $html = '';
+	my $template_data_ref_blocks = {};
+	my @blocks;
 
 	foreach my $block_ref (@{$blocks_ref}) {
-		$html .= "
-<div class=\"block\">
-<h3 class=\"block_title\">$block_ref->{title}</h3>
-<div class=\"block_content\">
-$block_ref->{content}
-</div>
-</div>
-";
-		if ((defined $block_ref->{id}) and ($block_ref->{id} eq 'my_block')) {
-			$html .= "<!-- end off canvas blocks for small screens -->\n";
-		}
-
+		push(@blocks, {
+			block_ref_content => $block_ref->{content},
```

Comment:
```suggestion
			content => $block_ref->{content},
```

We can remove the block_ref_ part from the name.

Reply:
In fact you don't even need to recreate a @blocks array: we already have one in $request_ref->{blocks_ref}

so you can do :

my $template_data_ref->{blocks} = $request_ref->{blocks_ref}